### PR TITLE
Mic Test Overlay

### PR DESCRIPTION
**Pull request by adrienntindall**
Follows from #311

### CODE REVIEW

File: src/overlay_mic_test.c [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+    u32 unk_3FF8;
+} UnkStruct_27E0000;
+
+void ov62_021E5CC8(void *data){
+    MicTestData *micTest = data;
+    NNS_GfdDoVramTransfer();
+    ov62_021E5FC4(micTest);
+    ov62_021E60D4(&micTest->unkB8);
+    UnkStruct_27E0000 *unkStruct = (UnkStruct_27E0000 *)(0x27e0000);
+    unkStruct->unk_3FF8 |= 1;
```
Comment (tgsm) [inline src/overlay_mic_test.c:227]:
```suggestion
    OS_SetIrqCheckFlag(OS_IE_VBLANK);
```

Comment (adrienntindall) [reply src/overlay_mic_test.c:227]: My hero

**Comment by PikalaxALT** [comment]
# Stale pull request warning

This pull request has had no activity for 60 days and has been marked stale. It will be closed after a further 30 days of inactivity.

**Comment by red031000** [comment]
this is literally a draft PR
draft PRs aren't subject to stale, I thought that was the entire idea

File: .githooks/pre-commit [hunk 2]
```diff
@@ -1,9 +0,0 @@
-#!/bin/sh
```
Comment (PikalaxALT) [inline .githooks/pre-commit:1]: Restore this

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -34,83 +33,70 @@ jobs:
...
-          mv NitroSDK-3_2-060901/include/nitro/specfiles/ARM7-TS.lcf.template $GITHUB_WORKSPACE/sub/
-          mv NitroSDK-3_2-060901/include/nitro/specfiles/ARM9-TS.lcf.template $GITHUB_WORKSPACE/
-          mv NitroSDK-3_2-060901/include/nitro/specfiles/mwldarm.response.template $GITHUB_WORKSPACE/
-        working-directory: ~
-
-      - name: Set Swap Space
-        uses: pierotofy/set-swap-space@master
-        with:
-          swap-size-gb: 19
+      - name: Build ROM
```
Comment (PikalaxALT) [inline .github/workflows/build.yml:40]: Is this from the rejected build-nonmatching PR? If so, please rever

File: include/constants/items.h [hunk 4]
```diff
@@ -389,7 +389,7 @@
 #define FIRST_MULCH_IDX  ITEM_GROWTH_MULCH
 #define LAST_MULCH_IDX   ITEM_GOOEY_MULCH
 #define NUM_MULCHES      (LAST_MULCH_IDX - FIRST_MULCH_IDX + 1)
-#define MULCH_IDX(mulch) ((mulch) - FIRST_MULCH_IDX + 1)
+#define MULCH_IDX(mulch) ((mulch)-FIRST_MULCH_IDX + 1)
```
Comment (PikalaxALT) [inline include/constants/items.h:392]: ???

File: include/constants/items.h [hunk 5]
```diff
@@ -506,7 +506,7 @@
 #define FIRST_BERRY_IDX  ITEM_CHERI_BERRY
 #define LAST_BERRY_IDX   ITEM_ROWAP_BERRY
 #define NUM_BERRIES      (LAST_BERRY_IDX - FIRST_BERRY_IDX + 1)
-#define BERRY_IDX(berry) ((berry) - FIRST_BERRY_IDX + 1)
+#define BERRY_IDX(berry) ((berry)-FIRST_BERRY_IDX + 1)
```
Comment (PikalaxALT) [inline include/constants/items.h:509]: ???

File: include/constants/std_script.h [hunk 6]
```diff
@@ -79,11 +79,11 @@
 #define std_apricorn_tree  2800
 
 #define _std_npc_trainer     3000
-#define std_trainer(trainer) ((trainer) - FIRST_TRAINER_INDEX + _std_npc_trainer)
+#define std_trainer(trainer) ((trainer)-FIRST_TRAINER_INDEX + _std_npc_trainer)
```
Comment (PikalaxALT) [inline include/constants/std_script.h:82]: etc

File: src/alph_puzzle.c [hunk 7]
```diff
@@ -195,10 +195,10 @@ static void Task_AlphPuzzle_WaitDropCursorAnimOnQuit(SysTask *task, void *_data)
 
 // Used to detect hovering over corners
 static const u8 sCornerCoords[4][2] = {
-    { 0, 0 },
-    { 5, 0 },
-    { 0, 5 },
-    { 5, 5 },
+    {0,  0},
```
Comment (PikalaxALT) [inline src/alph_puzzle.c:198]: clang-format, go home, you're drunk. and this is coming from someone who is drunk and ain't sobering up anytime soon.

File: src/data/fieldmap/hidden_items.h [hunk 8]
```diff
@@ -3,236 +3,236 @@
...
-    { ITEM_ZINC,         1, 0, 0, 119 },
-    { ITEM_FULL_HEAL,    1, 0, 0, 120 },
-    { ITEM_PROTEIN,      1, 0, 0, 221 },
-    { ITEM_MAX_REVIVE,   1, 0, 0, 121 },
-    { ITEM_ULTRA_BALL,   1, 0, 0, 123 },
-    { ITEM_NUGGET,       1, 0, 0, 124 },
-    { ITEM_DEEPSEATOOTH, 1, 0, 0, 38  },
-    { ITEM_DEEPSEASCALE, 1, 0, 0, 229 },
-    { ITEM_PP_MAX,       1, 0, 0, 230 },
+    {ITEM_POTION,        1, 0, 0, 0  },
```
Comment (PikalaxALT) [inline src/data/fieldmap/hidden_items.h:6]: my tea's gone cold i'm wonderin' why i got out of bed at all

File: src/easy_chat.c [hunk 9]
```diff
@@ -54,12 +54,12 @@ static const u16 sNarcMsgBanks[EC_GROUP_MAX] = {
 };
 
 static const u8 sLanguageToGreetingMap[6][2] = {
-    { LANGUAGE_JAPANESE, msg_0287_konnichiwa },
-    { LANGUAGE_ENGLISH,  msg_0287_hello      },
-    { LANGUAGE_FRENCH,   msg_0287_bonjour    },
-    { LANGUAGE_ITALIAN,  msg_0287_ciao       },
-    { LANGUAGE_GERMAN,   msg_0287_hallo      },
-    { LANGUAGE_SPANISH,  msg_0287_hola       },
+    {LANGUAGE_JAPANESE, msg_0287_konnichiwa},
```
Comment (PikalaxALT) [inline src/easy_chat.c:57]: my cold's gone tea i'm why wonderin' f\*\*\* me in the a\*\*

**Review by PikalaxALT — CHANGES_REQUESTED** [review]
i haven't even gotten to your new stuff but apparently clang-format is drunker than i've ever been

File: .githooks/pre-commit [hunk 10]
```diff
@@ -1,9 +1,9 @@
...
-# Redirect output to stderr.
-exec 1>&2
-
-# Run clang-format on staged files; abort the commit if any files are changed
-if ! git clang-format --extensions c,cpp,h,hpp ; then
-    echo "linting made changes to source files; aborting commit"
-    exit 1
-fi
+#!/bin/sh
```
Comment (PikalaxALT) [inline .githooks/pre-commit:1]: Fix line endings? should be \n, not \r\n

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

File: .githooks/pre-commit [hunk 11]
```diff
@@ -6,4 +6,4 @@ exec 1>&2
 if ! git clang-format --extensions c,cpp,h,hpp ; then
     echo "linting made changes to source files; aborting commit"
     exit 1
-fi
+fi
```
Comment (PikalaxALT) [inline .githooks/pre-commit:9]: Restore newline at EOF

File: .github/workflows/build-rom/action.yml [hunk 12]
```diff
@@ -0,0 +1,61 @@
+name: 'Build ROM'
```
Comment (PikalaxALT) [inline .github/workflows/build-rom/action.yml:1]: Still needed?

File: src/overlay_mic_test.c (hunk 1 above)
Comment (PikalaxALT) [reply src/overlay_mic_test.c:472]: Can you implement this?

File: src/overlay_mic_test.c [hunk 13]
```diff
@@ -0,0 +1,868 @@
...
+        }
+    }
+}
+
+static int MicTest_CheckTouchscreenInput() {
+    return TouchscreenHitbox_FindRectAtTouchNew(&sMicTestTouchscreenHitboxes[0]);
+}
+
+static int MicTest_CheckReturn(MicTestData *data) {
+    if (MicTest_CheckTouchscreenInput() == TS_HITBOX_MIC_TEST_RETURN || (gSystem.newKeys & 2)) {
```
Comment (PikalaxALT) [inline src/overlay_mic_test.c:864]:
```suggestion
    if (MicTest_CheckTouchscreenInput() == TS_HITBOX_MIC_TEST_RETURN || (gSystem.newKeys & PAD_BUTTON_B)) {
```

File: src/overlay_mic_test.c [hunk 14]
```diff
@@ -0,0 +1,868 @@
...
+
+    s16 x       = data->unk8.x >> 12;
+    s16 y       = data->unk8.y >> 12;
+    data->unk20 = data->unk20 + data->unk24;
+
+    while (data->unk20 > FX32_CONST(360)) {
+        data->unk20 -= FX32_CONST(360);
+    }
+
+    x += GF_SinDegNoWrap(data->unk20 >> 12) * data->unk28 >> 12;
```
Comment (PikalaxALT) [inline src/overlay_mic_test.c:839]:
```suggestion
    x += GF_SinDegNoWrap(data->unk20 >> FX32_SHIFT) * data->unk28 >> FX32_SHIFT;
```

File: src/overlay_mic_test.c [hunk 15]
```diff
@@ -0,0 +1,868 @@
...
+        if (--data->unk4 == 0) {
+            UnkImageStruct_SetSpriteVisibleFlag(data->unk0, TRUE);
+        }
+        return;
+    }
+
+    VEC_Add(&data->unk8, &data->unk14, &data->unk8);
+
+    s16 x       = data->unk8.x >> 12;
+    s16 y       = data->unk8.y >> 12;
```
Comment (PikalaxALT) [inline src/overlay_mic_test.c:832]:
```suggestion
    s16 x       = data->unk8.x >> FX32_SHIFT;
    s16 y       = data->unk8.y >> FX32_SHIFT;
```

File: src/overlay_mic_test.c [hunk 16]
```diff
@@ -0,0 +1,868 @@
...
+            args = &micTest->unk24[i];
+            flag = micTest->unk8[i + 4];
+            break;
+        }
+    }
+
+    if (args != NULL) {
+        s32 rand1 = (LCRandom() * 0x1002 / 0x10000) + 0x1333;
+        s32 rand2 = (LCRandom() * 0x3E / 0x10000) + 0x3C;
+        s32 rand3 = (LCRandom() * 0x10 / 0x10000) + 6;
```
Comment (PikalaxALT) [inline src/overlay_mic_test.c:787]: Resolve to base 10

File: src/overlay_mic_test.c [hunk 17]
```diff
@@ -0,0 +1,868 @@
...
+
+    if (volume <= 0x8C) {
+        ret = 0;
+    } else if (volume >= 0x8D && volume <= 0x98) {
+        ret = 1;
+    } else if (volume >= 0x99 && volume <= 0xA6) {
+        ret = 2;
+    } else if (volume >= 0xA7 && volume <= 0xBB) {
+        ret = 3;
+    } else if (volume >= 0xBC) {
```
Comment (PikalaxALT) [inline src/overlay_mic_test.c:755]: Resolve to base 10

File: src/overlay_mic_test.c [hunk 18]
```diff
@@ -0,0 +1,868 @@
...
+
+static void ov62_021E61AC(MicTestInput *input, HeapID heapId, MICCallback a2, MicTestData *micTest) {
+    Sys_SetSleepDisableFlag(8);
+
+    void *data = AllocFromHeap(heapId, 0x120);
+    MI_CpuFill8(data, 0, 0x120);
+    input->unk1C = data;
+
+    input->mic.type          = MIC_SAMPLING_TYPE_8BIT;
+    input->mic.buffer        = (void *)((((u32)data) + 0x1F) & ~0x1F); //??????
```
Comment (PikalaxALT) [inline src/overlay_mic_test.c:620]: this is alignment to a 32-byte boundary

File: src/overlay_mic_test.c [hunk 19]
```diff
@@ -0,0 +1,868 @@
...
+}
+
+static void ov62_021E60D4(MicTestSub_B8 *a0) {
+    if (a0->bgConfig != NULL) {
+        DoScheduledBgGpuUpdates(a0->bgConfig);
+    }
+}
+
+static void ov62_021E60E4(MicTestSub_B8 *a0, HeapID heapId) {
+    a0->msgData = NewMsgDataFromNarc(MSGDATA_LOAD_LAZY, NARC_msgdata_msg, 0xea, heapId);
```
Comment (PikalaxALT) [inline src/overlay_mic_test.c:589]:
```suggestion
    a0->msgData = NewMsgDataFromNarc(MSGDATA_LOAD_LAZY, NARC_msgdata_msg, NARC_msg_msg_00000234_bin, heapId);
```

File: src/overlay_mic_test.c [hunk 20]
```diff
@@ -0,0 +1,868 @@
...
+    for (u32 i = 0; i < 5; i++) {
+        InitBgFromTemplate(a0->bgConfig, ov62_021E6728[i].unk1C, &ov62_021E6728[i].bgTemplate, ov62_021E6728[i].unk1D);
+        BgClearTilemapBufferAndCommit(a0->bgConfig, ov62_021E6728[i].unk1C);
+        BG_ClearCharDataRange(ov62_021E6728[i].unk1C, 32, 0, heapId);
+    }
+}
+
+static void ov62_021E6024(MicTestSub_B8 *a0) {
+    for (u32 i = 0; i < 5; i++) {
+        FreeBgTilemapBuffer(a0->bgConfig, ov62_021E6728[i].unk1C);
```
Comment (PikalaxALT) [inline src/overlay_mic_test.c:564]: unk1C could be documented

File: src/overlay_mic_test.c [hunk 21]
```diff
@@ -0,0 +1,868 @@
...
+
+    SpriteRenderer_LoadPlttResObjFromNarcId(micTest->spriteRenderer, micTest->gfxHandler, NARC_a_1_7_6, 18, 0, 1, NNS_G2D_VRAM_TYPE_2DSUB, 0xD03);
+    SpriteRenderer_LoadCellResObjFromNarcId(micTest->spriteRenderer, micTest->gfxHandler, NARC_a_1_7_6, 20, 0, 0xD02);
+    SpriteRenderer_LoadAnimResObjFromNarcId(micTest->spriteRenderer, micTest->gfxHandler, NARC_a_1_7_6, 21, 0, 0xD04);
+    SpriteRenderer_LoadCharResObjFromNarcId(micTest->spriteRenderer, micTest->gfxHandler, NARC_a_1_7_6, 19, 0, NNS_G2D_VRAM_TYPE_2DSUB, 0xD01);
+
+    SpriteRenderer_LoadPlttResObjFromNarcId(micTest->spriteRenderer, micTest->gfxHandler, NARC_a_1_7_6, 14, 0, 1, NNS_G2D_VRAM_TYPE_2DMAIN, 0xC03);
+    SpriteRenderer_LoadCellResObjFromNarcId(micTest->spriteRenderer, micTest->gfxHandler, NARC_a_1_7_6, 16, 0, 0xC02);
+    SpriteRenderer_LoadAnimResObjFromNarcId(micTest->spriteRenderer, micTest->gfxHandler, NARC_a_1_7_6, 17, 0, 0xC04);
+    SpriteRenderer_LoadCharResObjFromNarcId(micTest->spriteRenderer, micTest->gfxHandler, NARC_a_1_7_6, 15, 0, NNS_G2D_VRAM_TYPE_2DMAIN, 0xC01);
```
Comment (PikalaxALT) [inline src/overlay_mic_test.c:520]: Stretch goal: consider unpacking NARC_a_1_7_6

File: src/overlay_mic_test.c [hunk 22]
```diff
@@ -0,0 +1,868 @@
...
+        if (IsPaletteFadeFinished()) {
+            (*state) = 2;
+        }
+        break;
+    case 2:
+        MicTest_SetTask(taskMan, MicTestTask_End);
+        break;
+    default:
+        GF_ASSERT(FALSE);
+    }
```
Comment (PikalaxALT) [inline src/overlay_mic_test.c:425]: state consts?

ditto above

File: src/overlay_mic_test.c [hunk 23]
```diff
@@ -0,0 +1,868 @@
...
+    u8 bgId;
+    u8 x;
+    u8 y;
+    u8 width;
+    u8 height;
+    u8 palNum;
+    u16 baseTile;
+    u8 fontId;
+    u32 color;
+} ov62_021E66F8[3] = {
```
Comment (PikalaxALT) [inline src/overlay_mic_test.c:247]: anonymous structs are discouraged where feasible

File: src/overlay_mic_test.c [hunk 24]
```diff
@@ -0,0 +1,868 @@
...
+    { .bgId = 0, .x = 1,  .y = 1,  .width = 9,  .height = 3, .palNum = 0, .baseTile = 1,    .fontId = 0, .color = 0xF0200 },
+    { .bgId = 4, .x = 5,  .y = 8,  .width = 22, .height = 5, .palNum = 0, .baseTile = 1,    .fontId = 0, .color = 0xF0200 },
+    { .bgId = 4, .x = 24, .y = 21, .width = 6,  .height = 2, .palNum = 0, .baseTile = 0x6F, .fontId = 4, .color = 0xE0F00 },
```
Comment (PikalaxALT) [inline src/overlay_mic_test.c:250]: use MAKE_TEXT_COLOR macro

File: src/overlay_mic_test.c [hunk 25]
```diff
@@ -0,0 +1,868 @@
...
+     .unk1F      = 0,
+     },
+};
+
+static const UnkTemplate_0200D748 ov62_021E67C8[5] = {
+    { .x = 0x40, .y = 0x60, .z = 0, .animation = 0, .spritePriority = 0, .pal = 0, .vram = NNS_G2D_VRAM_TYPE_2DMAIN, .resIdList = { 0xA01, 0xA03, 0xA02, 0xA04, 0, 0 }, .bgPriority = 0, .vramTransfer = 0 },
+    { .x = 0xC0, .y = 0x60, .z = 0, .animation = 0, .spritePriority = 0, .pal = 0, .vram = NNS_G2D_VRAM_TYPE_2DMAIN, .resIdList = { 0xA01, 0xA03, 0xA02, 0xA04, 0, 0 }, .bgPriority = 0, .vramTransfer = 0 },
+    { .x = 0x80, .y = 0x60, .z = 0, .animation = 0, .spritePriority = 0, .pal = 0, .vram = NNS_G2D_VRAM_TYPE_2DMAIN, .resIdList = { 0xB01, 0xB03, 0xB02, 0xB04, 0, 0 }, .bgPriority = 0, .vramTransfer = 0 },
+    { .x = 0xD8, .y = 0xB0, .z = 0, .animation = 0, .spritePriority = 0, .pal = 0, .vram = NNS_G2D_VRAM_TYPE_2DSUB,  .resIdList = { 0xD01, 0xD03, 0xD02, 0xD04, 1, 0 }, .bgPriority = 0, .vramTransfer = 0 },
+    { .x = 0,    .y = 0,    .z = 0, .animation = 0, .spritePriority = 0, .pal = 0, .vram = NNS_G2D_VRAM_TYPE_2DMAIN, .resIdList = { 0xC01, 0xC03, 0xC02, 0xC04, 0, 0 }, .bgPriority = 0, .vramTransfer = 0 },
```
Comment (PikalaxALT) [inline src/overlay_mic_test.c:193]: if using designated initializers, each field should be on its own line. exception is unions (i.e. TouchscreenHitbox, below)

File: src/overlay_mic_test.c [hunk 26]
```diff
@@ -0,0 +1,868 @@
...
+            .areaOver   = 0,
+            .dummy      = 0,
+            .mosaic     = 0 },
+     .unk1C = 0,
+     .unk1D = 0,
+     .unk1E = 0,
+     .unk1F = 0,
+     },
+    {
+     .bgTemplate = { .x = 0, .y = 0, .bufferSize = 0x800, .baseTile = 0, .size = 1, .colorMode = 0, .screenBase = 0x1E, .charBase = 1, .bgExtPltt = 0, .priority = 1, .areaOver = 0, .dummy = 0, .mosaic = 0 },
```
Comment (PikalaxALT) [inline src/overlay_mic_test.c:159]: make these BgTemplates consistent

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

File: src/overlay_mic_test.c (hunk 14 above)
Comment (tgsm) [reply src/overlay_mic_test.c:839]: Can also use FX_Whole

File: src/overlay_mic_test.c (hunk 15 above)
Comment (tgsm) [reply src/overlay_mic_test.c:832]: ditto

File: src/overlay_mic_test.c [hunk 27]
```diff
@@ -0,0 +1,868 @@
...
+} ov62_021E66F8[3] = {
+    { .bgId = 0, .x = 1,  .y = 1,  .width = 9,  .height = 3, .palNum = 0, .baseTile = 1,    .fontId = 0, .color = 0xF0200 },
+    { .bgId = 4, .x = 5,  .y = 8,  .width = 22, .height = 5, .palNum = 0, .baseTile = 1,    .fontId = 0, .color = 0xF0200 },
+    { .bgId = 4, .x = 24, .y = 21, .width = 6,  .height = 2, .palNum = 0, .baseTile = 0x6F, .fontId = 4, .color = 0xE0F00 },
+};
+
+const OVY_MGR_TEMPLATE gApplication_MicTest = { MicTest_Init, MicTest_Main, MicTest_Exit, FS_OVERLAY_ID_NONE };
+
+static BOOL MicTest_Init(OVY_MANAGER *overlayMan, int *state) {
+    CreateHeap(HEAP_ID_3, HEAP_ID_MIC_TEST, 3 << 0x10);
```
Comment (tgsm) [inline src/overlay_mic_test.c:256]:
```suggestion
    CreateHeap(HEAP_ID_3, HEAP_ID_MIC_TEST, 0x30000);
```

File: src/overlay_mic_test.c [hunk 28]
```diff
@@ -0,0 +1,868 @@
...
+static void ov62_021E61AC(MicTestInput *input, HeapID heapId, MICCallback a2, MicTestData *micTest) {
+    Sys_SetSleepDisableFlag(8);
+
+    void *data = AllocFromHeap(heapId, 0x120);
+    MI_CpuFill8(data, 0, 0x120);
+    input->unk1C = data;
+
+    input->mic.type          = MIC_SAMPLING_TYPE_8BIT;
+    input->mic.buffer        = (void *)((((u32)data) + 0x1F) & ~0x1F); //??????
+    input->mic.size          = (1 << 8);
```
Comment (tgsm) [inline src/overlay_mic_test.c:621]: 256, or 0x100

File: src/overlay_mic_test.c [hunk 29]
```diff
@@ -0,0 +1,868 @@
...
+static void MicTestTaskMan_Finish(MicTestTaskManager *taskMan) {
+    taskMan->isFinished = TRUE;
+}
+
+static void MicTestTask_FadeIn(MicTestTaskManager *taskMan, u32 *state) {
+    MicTestData *micTest = MicTestTaskMan_GetMicTestData(taskMan);
+
+    switch (*state) {
+    case 0:
+        BeginNormalPaletteFade(0, 1, 1, 0, 6, 1, micTest->heapId);
```
Comment (tgsm) [inline src/overlay_mic_test.c:343]:
```suggestion
        BeginNormalPaletteFade(0, 1, 1, RGB_BLACK, 6, 1, micTest->heapId);
```

File: src/overlay_mic_test.c [hunk 30]
```diff
@@ -0,0 +1,868 @@
...
+        GF_ASSERT(FALSE);
+    }
+}
+
+static void MicTestTask_FadeOut(MicTestTaskManager *taskMan, u32 *state) {
+    MicTestData *micTest = MicTestTaskMan_GetMicTestData(taskMan);
+
+    switch (*state) {
+    case 0:
+        BeginNormalPaletteFade(0, 0, 0, 0, 6, 1, micTest->heapId);
```
Comment (tgsm) [inline src/overlay_mic_test.c:412]:
```suggestion
        BeginNormalPaletteFade(0, 0, 0, RGB_BLACK, 6, 1, micTest->heapId);
```

**Review by tgsm — CHANGES_REQUESTED** [review]

File: src/overlay_mic_test.c (hunk 21 above)
Comment (adrienntindall) [reply src/overlay_mic_test.c:520]: Dont wanna do it this pr bc I dont wanna think about it anymore

File: src/overlay_mic_test.c [hunk 31]
```diff
@@ -0,0 +1,1023 @@
...
+    u32 unk_3FF8;
+} UnkStruct_27E0000;
+
+static void MicTest_VBlankIntrCB(void *data) {
+    MicTestData *micTest = data;
+    NNS_GfdDoVramTransfer();
+    ov62_021E5FC4(micTest);
+    ov62_021E60D4(&micTest->unkB8);
+    UnkStruct_27E0000 *unkStruct = (UnkStruct_27E0000 *)(0x27e0000);
+    unkStruct->unk_3FF8 |= 1;
```
Comment (tgsm) [inline src/overlay_mic_test.c:627]: This was never fixed

**Review by tgsm — APPROVED** [review]

**Review by PikalaxALT — APPROVED** [review]

**Pull request merged**